### PR TITLE
fix: callback logic + small fixex/optim

### PR DESCRIPTION
**Pull request by simon-something**
# Description

- Prevent potential permit2 DOS by frontrunning the call to permit()
- zeroForOne is reused instead of recomputed in _swap
- 
## Limitations & risks

*Are there any trade-off or new vulnarbility surface based on theses changes?*

# Check-list
- [ ] Tests are covering the new feature
- [ ] Code is [natspec'd](https://docs.soliditylang.org/en/v0.8.17/natspec-format.html)
- [ ] Code is [linted and formatted](https://docs.soliditylang.org/en/v0.8.17/style-guide.html)
- [ ] I have run the test locally (and they pass)
- [ ] I have rebased to the latest main commit (and tests still pass)

# Interactions
These changes will impact the following contracts:
- Directly:

- Indirectly:

### CODE REVIEW

File: src/JBSwapTerminal.sol [hunk 1]
```diff
@@ -709,7 +743,13 @@ contract JBSwapTerminal is JBPermissioned, Ownable, IJBTerminal, IJBPermitTermin
                 sigDeadline: allowance.sigDeadline
             }),
             signature: allowance.signature
-        }) {} catch {}
+        }) {} catch {
+            // Allowance already previously set?
+            (uint160 amount, uint48 expiration, uint48 nonce) = PERMIT2.allowance(msg.sender, token, address(this));
+            if(amount < allowance.amount || expiration < allowance.expiration || nonce < allowance.nonce) {
```
Comment (xBA5ED) [inline src/JBSwapTerminal.sol]: nit: I think this check isn't necessarily needed as it will revert when we call it to perform the transfer.

**Review by xBA5ED — DISMISSED** [review]
We need to fix the tests, but lgtm otherwise

File: src/JBSwapTerminal.sol (hunk 1 above)
Comment (mejango) [reply src/JBSwapTerminal.sol]: i dont think we have contract-space to add it to the jBMultiterminal. maybe lets remove it here too if its not needed.

Comment (mejango) [reply src/JBSwapTerminal.sol]: here's the audit context: https://github.com/Enigma-Dark/nana-swap-terminal/issues/10

Comment (mejango) [reply src/JBSwapTerminal.sol]: gonna bias towards removing. keep me posted if we need to add it back, i agree with the remove take

**Review by mejango — APPROVED** [review]

**Pull request merged**